scipy/integrate/quadpack.py

```diff
@@ -704,22 +704,21 @@ def nquad(func, ranges, args=None, opts=None, full_output=False):
     ----------
     func : {callable, scipy.LowLevelCallable}
         The function to be integrated. Has arguments of ``x0, ... xn``,
-        ``t0, tm``, where integration is carried out over ``x0, ... xn``, which
-        must be floats.  Function signature should be
-        ``func(x0, x1, ..., xn, t0, t1, ..., tm)``.  Integration is carried out
-        in order.  That is, integration over ``x0`` is the innermost integral,
-        and ``xn`` is the outermost.
+        ``t0, ... tm``, where integration is carried out over ``x0, ... xn``,
+        which must be floats.  Where ```t0, ... tm``` are extra arguments
+        passed in args.
+        Function signature should be ``func(x0, x1, ..., xn, t0, t1, ..., tm)``.
+        Integration is carried out in order.  That is, integration over ``x0``
+        is the innermost integral, and ``xn`` is the outermost.
 
         If the user desires improved integration performance, then `f` may
         be a `scipy.LowLevelCallable` with one of the signatures::
 
             double func(int n, double *xx)
             double func(int n, double *xx, void *user_data)
 
-        where ``n`` is the number of extra parameters and args is an array
-        of doubles of the additional parameters, the ``xx`` array contains the
-        coordinates. The ``user_data`` is the data contained in the
-        `scipy.LowLevelCallable`.
+        where ``n`` is the number of variables plus extra parameters.
+        The ``xx`` array contains the coordinates and ``user_data`` contains ???
```

Comment:
I'm not sure this is actually what's happening,

`n` seems to be number of variables + extra params
`xx` seems to be variables + extra params
`user_data` is always empty?


```
#include "stdio.h"
// same result with 
//double f(int n, double args[]) {
//double f(int n, double args[], void* user_data, void* k) {
double f(int n, double args[], void* user_data) {
    (void)args;
    printf("%i [%.1f %.1f %.1f %.1f] | [%.1f] \n",
        n,
        args[0], args[1], args[2], args[3],
        ((double*) user_data)[0]);

    return args[0] + args[1] + args[2];
}
```
`rm -f func.so; gcc -fPIC -shared func.c -o func.so`

```
import ctypes
from scipy.integrate import nquad
lib = ctypes.CDLL('/home/eights/Projects/scipy/func.so')
func = lib.f
func.restype = ctypes.c_double
func.argtypes = (ctypes.c_int, ctypes.c_double)
print(nquad(func, [[0, 1],  [1, 2], [2,3]], args=None))
print(nquad(func, [[0, 1],  [1, 2], [2,3]], args=[7]))
print(nquad(func, [[0, 1],  [1, 2], [2,3]], args=[11,13]))
```
```
3 [0.9 1.4 2.6 0.0] | [0.0] 
...
4 [0.9 1.4 2.6 7.0] | [0.0] 
...
5 [0.6 1.6 2.6 11.0] | [0.0] 
```

Reply:
You are here using the legacy interface: passing in a ctypes function, not a LowLevelCallable. user_data is null because of that.

The documentation does not currently describe the legacy interface.

Reply:
Thanks @pv, rereading the documentation I understand `user_data` I modified the docstring (and edited my comment above)